newt/pkg/bsp_package.go

```diff
@@ -103,14 +113,33 @@ func (bsp *BspPackage) resolveLinkerScriptSetting(
 			return nil, err
 		}
 
-		if path != "" {
+		if path == "autogenerated" {
+			path, err = bsp.getAutogeneratedLinkerScriptPath()
+			if err != nil {
+				return nil, util.PreNewtError(err,
+					"Could not resolve autogenerated linker script path for target \"%s\"",
+					bsp.yov.Pkg.Name())
+			}
+			paths = append(paths, path)
+		} else if path != "" {
 			paths = append(paths, path)
 		}
 	} else {
 		proj := interfaces.GetProject()
 
 		// Read each linker script from the list.
 		for _, val := range vals {
+			if val == "autogenerated" {
+				path, err := bsp.getAutogeneratedLinkerScriptPath()
+				if err != nil {
+					return nil, util.PreNewtError(err,
+						"Could not resolve autogenerated linker script path for target \"%s\"",
+						bsp.yov.Pkg.Name())
+				}
+				paths = append(paths, path)
+				continue
+			}
```

Comment:
autogenerated scripts should be only handled if single script is set to "autogenerated", here it should better just return an error